System.Compiler/Reader.cs

```diff
@@ -2129,13 +2129,28 @@ private TypeNode GetTypeFromTypeName(SerializedTypeName typeName)
         string lookupName = localInfo.Name;
         bool hasNoPDBName = String.IsNullOrEmpty(lookupName);
 #if !FxCop
-        string name = hasNoPDBName ? "local"+i : lookupName;
+        
+        // In some case it is very useful to know is the local has real name or
+        // the name was generated by the reader.
+        bool anonymous = false;
+        string name;
+
+        if (hasNoPDBName)
+        {
+            name = "local" + i;
+            anonymous = true;
+        }
+        else
+        {
+            name = lookupName;
+        }
 #else
         string name = hasNoPDBName ? "local$"+i : lookupName;
 #endif
         bool pinned = false;
         TypeNode locType = this.ParseTypeSignature(sigReader, ref pinned);
-        Local loc = new Local(Identifier.For(name), locType);
+        
+        Local loc = new Local(Identifier.For(name), locType) {Anonymous = anonymous};
```

Comment:
:bulb: I don't like the inconsistency of using an object initializer for one property and separate statements for several more. Since the old code wasn't using an object initializer, I'd prefer to just match what already exists and use a new assignment statement.